### PR TITLE
create a story

### PR DESCRIPTION
**Pull request by martcamelia**

### CODE REVIEW

File: Project/Views/UserProfile/_Story.cshtml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿@model Project.ViewModels.Story.StoryVM
+<div class="card-body">
+    <div class="row">
+        <div class="col-8"><h3>Add Story</h3></div>
+    </div>
+        @using (Html.BeginForm("AddStory", "UserProfile", new { }, FormMethod.Post, new { @class = "form-horizontal", id = "addstory-form", role = "form" }))
+        {
+            @Html.AntiForgeryToken()
+
+            @Html.HiddenFor(x => x.Id)
```
Comment (dimarobert) [inline Project/Views/UserProfile/_Story.cshtml]: If this is an add only form, the Id is not needed. It will be automatically generated on insert.
Will this form be used for edits also?

File: StoryDomain/Models/Story.cs [hunk 2]
```diff
@@ -21,7 +21,9 @@ public class Story : ObjectWithState, IEntity<int> {
 
         public StoryType Type { get; set; }
 
-        public virtual UserInfo User { get; set; }
+        public DateTime? Date { get; set; }
+
+        //public virtual UserInfo User { get; set; }
```
Comment (dimarobert) [inline StoryDomain/Models/Story.cs]: Why was this removed.

File: Project/Controllers/UserProfileController.cs [hunk 3]
```diff
@@ -188,35 +213,41 @@ public class UserProfileController : Controller {
...
+            {
                 return PartialView("_AjaxValidation", "Required story fields were not filled in.");
             }
 
             var storyModel = Mapper.Map<Story>(story);
-            storyModel.Id = 0;
+
             storyModel.State = Core.Models.ModelState.Added;
+            storyModel.Date = DateTime.Now;
+            UserInfo user = await userService.FindUserByIdAsync(storyModel.UserId);
```
Comment (dimarobert) [inline Project/Controllers/UserProfileController.cs]: This is not used. Also, the userId should be retrieved with the userService.GetUserId() because the story owner should be the currently logged in user. Setting the NewStory.UserId in the Index() method is also redundant.

File: Project/Views/UserProfile/_Story.cshtml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            @Html.HiddenFor(x => x.UserId)
```
Comment (dimarobert) [inline Project/Views/UserProfile/_Story.cshtml]: UserId will be determined based on the currently logged in user.

File: Project/Views/UserProfile/_Story.cshtml (hunk 1 above)
Comment (martcamelia) [reply Project/Views/UserProfile/_Story.cshtml]: That's what I initially thought. But I don't think we can use it.

File: StoryDomain/Models/Story.cs (hunk 2 above)
Comment (martcamelia) [reply StoryDomain/Models/Story.cs]: Because I cant save a story, altough I set the User with the method FindUserById, it gives an error.

File: Project/Controllers/UserProfileController.cs (hunk 3 above)
Comment (martcamelia) [reply Project/Controllers/UserProfileController.cs]: I put that before deleting the User from Story model.
